src/Submissions/Api/Services/Directory/MaterialTypeService.cs

```diff
@@ -37,6 +38,15 @@ public Task<IEnumerable<OntologyTerm>> GetMaterialTypeExtractionProcedures(int i
             .Where(x => x.DisplayOnDirectory || !onlyDisplayable));
 
 
+
+    public async Task<int> GetMaterialTypeMaterialDetailCount(int id)
+        => await _db.MaterialDetails.CountAsync(x => x.MaterialTypeId == id);
+
+    public async Task<bool> IsMaterialTypeAssigned(int id)
+        => await _db.OntologyTerms
+            .Include(x => x.MaterialTypes)
+            .Where(x => x.SnomedTag != null && x.SnomedTag.Value == SnomedTags.ExtractionProcedure)
+            .AnyAsync(x => x.MaterialTypes.Any(y => y.Id == id));
```

Comment:
```suggestion
            .AnyAsync(x => x.MaterialTypes.Any(y => y.Id == id));
            
```

(line between methods)